src/System.Diagnostics.Activity/Activity.cs

```diff
@@ -132,6 +110,52 @@ public override string ToString()
             return $"operation: {OperationName}, Id={Id}, context: {{{dictionaryToString(Baggage)}}}, tags: {{{dictionaryToString(Tags)}}}";
         }
 
+        public static bool CurrentEnabled { get; set; } = true;
+
+        public static void SetCurrent(Activity newActivity)
+        {
+            if (CurrentEnabled)
+            {
+                _current.Value = newActivity;
+            }
+        }
+
+        #region private 
+        private string GenerateId()
+        {
+            string ret;
+            if (Parent != null)
+#if DEBUG 
+                ret = Parent.Id + "/" + OperationName + "_" + Interlocked.Increment(ref Parent._currentChildId);
+#else           // To keep things short, we drop the operation name 
+                ret = Parent.Id + "/" + Interlocked.Increment(ref Parent._currentChildId);
+#endif
+            else
+            {
+                if (_uniqPrefix == null)
+                {
+                    // Here we make an ID to represent the Process/AppDomain.   Ideally we use process ID but 
+                    // it is unclear if we have that ID handy.   Currently we use low bits of high freq tick 
+                    // as a unique random number (which is not bad, but loses randomness for startup scenarios).  
+                    int uniqNum = (int)Stopwatch.GetTimestamp();
+                    string uniqPrefix = $"//{Environment.MachineName}_{uniqNum:x}/";
+                    Interlocked.CompareExchange(ref _uniqPrefix, uniqPrefix, null);
+                }
+#if DEBUG
+                ret = _uniqPrefix + OperationName + "_" + Interlocked.Increment(ref _currentRootId);
+#else           // To keep things short, we drop the operation name 
+                ret = _uniqPrefix + Interlocked.Increment(ref _currentRootId);
+#endif 
+            }
+            // Useful place to place a conditional breakpoint.  
+            return ret;
+        }
+
+        // Used to generate an ID 
+        int _currentChildId;            // A unique number for all children of this activity.  
+        static int _currentRootId;      // A unique number inside the appdomain.
```

Comment:
We now have random hierarchical ids, this is awesome!

I think _currentChildId and _currentRootId should be long.
If int is used, for heavily load service, id collision could technically happen. E.g. instance gets 1000 requests per sec, ids would repeat after ~50 days.